client/app/src/main/java/com/microsoft/research/karya/ui/registration/ProfilePictureFragment.kt

```diff
@@ -0,0 +1,156 @@
+            mainProfilePictureIv.isClickable = false
+            rotateRightIb.isClickable = false
+            profilePictureNextIv.visibility = View.INVISIBLE
+            submitProfilePicture()
+        }
+
+        rotateRightIb.setOnClickListener { rotateRight() }
+
+        disableRotateButton()
+
+    }
+
+    /**
+     * On resume, play assistant audio if a profile pic is not already taken
+     */
+    override fun onResume() {
+        super.onResume()
+        if (!::profilePic.isInitialized) {
+            baseActivity.onAssistantClick()
+        }
+    }
+
+    /**
+     * Handle camera completion
+     */
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        // Profile picture returned by the camera
+        if (requestCode == REQUEST_IMAGE_CAPTURE && resultCode == AppCompatActivity.RESULT_OK) {
+            profilePic = data?.extras?.get("data") as Bitmap
+            mainProfilePictureIv.setBackgroundResource(0)
+            ImageUtils.loadImageBitmap(requireActivity(), profilePic, mainProfilePictureIv)
+            enableRotateButton()
+        }
+    }
+
+    /**
+     * Initiate the camera to capture profile picture
+     */
+    private fun getProfilePicture() {
+        Intent(MediaStore.ACTION_IMAGE_CAPTURE).also { takePictureIntent ->
+            takePictureIntent.resolveActivity(requireActivity().packageManager)?.also {
+                startActivityForResult(takePictureIntent, REQUEST_IMAGE_CAPTURE)
+            }
+        }
+    }
+
+    /**
+     * Handle submit profile picture click.
+     */
+    private fun submitProfilePicture() {
+        if (::profilePic.isInitialized) {
+            WorkerInformation.profile_picture = profilePic
+            val imageFolder = requireActivity().getDir("profile_picture", AppCompatActivity.MODE_PRIVATE).path
+            val fileName = "pp.png"
+            val out = FileOutputStream("$imageFolder/$fileName")
+            profilePic.compress(Bitmap.CompressFormat.PNG, 100, out)
+        }
+        startActivity(Intent(activity, SelectGenderActivity::class.java))
+    }
+
+
+    /**
+     * Disable rotate button
+     */
+    private fun disableRotateButton() {
+        rotateRightIb.visibility = View.INVISIBLE
+    }
+
+    /**
+     * Enable rotate button
+     */
+    private fun enableRotateButton() {
+        rotateRightIb.visibility = View.VISIBLE
+        rotateRightIb.isClickable = true
+    }
+
+    /**
+     * Rotate right
+     */
+    private fun rotateRight() {
+        if (::profilePic.isInitialized) {
+            val matrix = Matrix()
+            matrix.postRotate(90.toFloat())
+            profilePic = Bitmap.createBitmap(
+                profilePic, 0, 0, profilePic.width, profilePic.height,
+                matrix, true
+            )
+            ImageUtils.loadImageBitmap(requireActivity(), profilePic, mainProfilePictureIv)
+        }
+    }
+
```

Comment:
I believe this function can be improved by removing all the side effects. We can do something along the lines of

```
fun Bitmap.rotateRight(): Bitmap {
	val matrix = Matrix()
    matrix.postRotate(90.toFloat())
    
    return Bitmap.createBitmap(this, 0, 0, width, height, matrix, true)
}
```